clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -143,20 +144,36 @@ record_filter_tool_create(const std::string &output_dir, uint64_t stop_timestamp
                 new dynamorio::drmemtrace::trim_filter_t(trim_before_timestamp,
                                                          trim_after_timestamp)));
     }
+    if (encoding_filter_enabled) {
+        filter_funcs.emplace_back(
+            std::unique_ptr<dynamorio::drmemtrace::record_filter_t::record_filter_func_t>(
+                new dynamorio::drmemtrace::encoding_filter_t()));
+    }
+
     // TODO i#5675: Add other filters.
 
-    return new dynamorio::drmemtrace::record_filter_t(output_dir, std::move(filter_funcs),
-                                                      stop_timestamp, verbose);
+    /* If we are changing the encoding of trace_entry_t, we will be generating
+     * discrepancies between encoding size and instruction length.  So, we need to tell
+     * reader_t, which here comes in the form of memref_counter_t, to ignore such
+     * discrepancies.  Note that simulators that deal with these filtered traces will
```

Comment:
What about reader_t used by memref_t analysis tools on the final trace: will we get errors on any analysis tool use?

Reply:
Solved with changing the trace file_type.